client/components/jetpack/card/jetpack-product-card/upgrade-nudge.tsx

```diff
@@ -0,0 +1,130 @@
+	className?: string;
+	currencyCode: string;
+	discountedPrice?: number;
+	features?: FeaturesItem[];
+	onUpgradeClick: () => void;
+	originalPrice: number;
+	productType?: string;
+	selectorProduct: SelectorProduct;
+};
+
+const UpgradeNudge = ( {
+	billingTimeFrame,
+	className,
+	currencyCode,
+	discountedPrice,
+	features = DEFAULT_UPGRADE_NUDGE_FEATURES,
+	onUpgradeClick,
+	originalPrice,
+	selectorProduct,
+	productType = 'Real-time',
+}: OwnProps ) => {
+	const translate = useTranslate();
+	const isDiscounted = isFinite( discountedPrice );
+
+	const storedPreference = useSelector( ( state ) =>
+		getPreference( state, JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS )
+	);
+
+	const { description, productSlug } = selectorProduct;
+
+	// Save dismiss to never show up the nudge again for this specific plan/product.
+	const dispatch = useDispatch();
+	const onCancelClick = useCallback( () => {
+		dispatch(
+			savePreference( JETPACK_OFFER_RESET_UPGRADE_NUDGE_DISMISS, {
+				...storedPreference,
+				[ productSlug ]: Date.now(),
+			} )
+		);
+	}, [ dispatch, productSlug, storedPreference ] );
+
+	// If this nudge was dismissed in the past don't show the nudge.
+	if ( storedPreference && storedPreference[ productSlug ] ) {
+		return null;
+	}
+
+	return (
+		<div className={ classNames( 'jetpack-product-card__nudge', className ) }>
+			<div className="jetpack-product-card__summary">
+				<div className="jetpack-product-card__nudge-heading">
+					<h3>
+						{ translate( 'Upgrade to' ) }
+						<br />
+						<span className="jetpack-product-card__nudge-product-type">{ productType }</span>
+					</h3>
+				</div>
+				<div className="jetpack-product-card__price">
+					<span className="jetpack-product-card__raw-price">
+						<PlanPrice
+							rawPrice={ originalPrice }
+							original={ isDiscounted }
+							currencyCode={ currencyCode }
+						/>
+						{ isDiscounted && (
+							<PlanPrice rawPrice={ discountedPrice } discounted currencyCode={ currencyCode } />
+						) }
+					</span>
+					<span className="jetpack-product-card__billing-time-frame">{ billingTimeFrame }</span>
+				</div>
+			</div>
+			<p className="jetpack-product-card__nudge-description">{ preventWidows( description ) }</p>
+
+			<Button className="jetpack-product-card__nudge-button" primary onClick={ onUpgradeClick }>
+				{ translate( 'Upgrade' ) }
+			</Button>
+
+			<ul className="jetpack-product-card__nudge-features">
+				{ features.map( ( feature, index ) => (
+					<li className="jetpack-product-card__nudge-feature" key={ index }>
+						<Gridicon icon="checkmark" />
+						<div className="jetpack-product-card__nudge-feature-desc">
+							<strong>{ feature.text }</strong>
+							<br />
+							{ feature.description }
+						</div>
+					</li>
+				) ) }
+			</ul>
+
+			<Button
+				className="jetpack-product-card__nudge-button jetpack-product-card__nudge-button--cancel"
+				borderless
+				onClick={ onCancelClick }
+			>
+				{ translate( 'No thanks' ) }
```

Comment:
Adding context would perhaps be helpful for translators?

Reply:
Sure. I always forget to do that. Thanks for the reminder.